src/js/events/search/SearchHistoryManager.js

```diff
@@ -0,0 +1,18 @@
+export class SearchHistoryManager {
+  constructor() {
+    this.history = JSON.parse(localStorage.getItem('searchHistory')) || [];
+  }
+
+  addSearch(value) {
+    if (this.isDuplicate(value)) {
+      return;
+    } else {
+      this.history.push(value);
+      localStorage.setItem('searchHistory', JSON.stringify(this.history));
+    }
```

Comment:
```suggestion
    if (this.isDuplicate(value)) {
      return;
    } 
    
    this.history.push(value);
    localStorage.setItem('searchHistory', JSON.stringify(this.history));
```

요것은 취향입니다만,

early return을 했다면 굳이 else를 쓸 필요는 없죵